cmd/titus-metadata-service/main.go

```diff
@@ -158,23 +189,43 @@ func main() {
 		/* Get the requisite configuration from environment variables */
 		listener := getListener(listenPort, listenerFd)
 
-		var err error
-		var container *titus.ContainerInfo
-		var signer *identity.Signer
-
-		if container, err = readTaskConfigFile(titusTaskInstanceID); err != nil {
-			log.Fatal(err)
+		mdscfg := types.MetadataServerConfiguration{
+			IAMARN:              iamARN,
+			TitusTaskInstanceID: titusTaskInstanceID,
+			Ipv4Address:         net.ParseIP(ipv4Address),
+			VpcID:               vpcID,
+			EniID:               eniID,
+			Region:              region,
+			Optimistic:          optimistic,
+			APIProtectEnabled:   apiProtectEnabled,
+		}
+		if parsedURL, err := url.Parse(backingMetadataServer); err == nil {
+			mdscfg.BackingMetadataServer = parsedURL
+		} else {
+			return cli.NewExitError(err.Error(), 1)
 		}
+
 		if metatronEnabled {
 			log.Info("Metatron enabled!")
-			if signer, err = identity.NewDefaultSigner(); err != nil {
-				log.Fatal(err)
+			if signer, err := identity.NewDefaultSigner(); err != nil {
+				log.WithError(err).Fatal("Cannot instantiate new default signer")
+				mdscfg.Signer = signer
+			}
+			if container, err := readTaskConfigFile(titusTaskInstanceID); err != nil {
+				log.WithError(err).Fatal("Cannot read container config file")
+				mdscfg.Container = container
 			}
 		}
-		ms := metadataserver.NewMetaDataServer(context.Background(), backingMetadataServer, iamARN, titusTaskInstanceID, ipv4Address, region, optimistic, container, signer)
+
+		if ipv6Addresses != "" {
```

Comment:
Technically EC2_IPV6s can have multiple IPs. I believe they get newline separated. We only care for the primary IP.

Reply:
Might be helpful to move this comment into the code :)